### ISSUE TITLE
Examing .set() alternatives -- move to ES6 getter and setter methods?

### ISSUE THREAD

**Issue by taylorbf** [nexus-js/ui#31]
From an API perspective, `slider1.value = 0.4` is more convenient than `slider1.val.value = 0.4; slider1.draw();` or `slider1.set( .... )`. 

Look into approaches for a simplified API with same core functionality and ease of development.

**Comment by sebpiq** [comment]
ES6 has very bad support at the moment. You should look into `Object.createProperty` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty it is part of ES5, therefore supported almost everywhere.

**Comment by adowty351** [comment]
I'm trying to set a select object programmatically.
The syntax from the slider.set example isn't working.
?

**Comment by taylorbf** [comment]
Hi @adowty351, you are right, thanks for catching that. Select is an exceptional widget in the library because it uses a select DOM element instead of a canvas, but I do want it to have a .set() method just like the others.

I will make this edit and close the issue when I post the update.

In the meantime (next 24 hrs) the following code should do it:

```
select1.canvas.value = "YourOptionText"
select1.change(select1.canvas)
```

**Comment by taylorbf** [comment]
@adowty351, I updated the master branch to address the issue you mentioned. 

This is the code to programmatically set a select object (here called `select1`):

```
select1.set({ text: "MyOption" })
```

**Comment by taylorbf** [comment]
Getters/setters implemented in Version 2: b03b8f6ea20e9d184e466b91c8475c065a676af1 

API now lets you set the value of a slider like this: `slider1.value = 0.5`
